lib/client-credentials.js

```diff
@@ -0,0 +1,87 @@
+'use strict'
+
+const Client = require('../lib/client')
+const Token = require('../lib/token')
+const Promise = require('bluebird')
+const uuid = require('uuid')
+
+module.exports = {
+  getAccessToken: function (bearerToken) {
+    var token = null
+
+    return Token.objects.get({
+      access_token: bearerToken
+    })
+    .then(function (_token) {
+      token = _token
+      return token.client
+    })
```

Comment:
It may be helpful to separate the "get a token" promise from the "get a client" promise, and return a promise that's dependent on both prior values:

```javascript
const getToken = Token.objects.get({access_token: bearerToken})
const getClient = getToken.get('client')
return Promise.join(
  getToken,
  getClient
).spread((token, client) => {
  return {
    accessToken: token.access_token,
    client,
    accessTokenExpiresAt: token.access_expires,
    refreshTokenExpiresAt: token.refresh_expires,
    user: {
      email: token.user_email
    }
  }
})
```